algorithms/src/snark/varuna/varuna.rs

```diff
@@ -113,7 +113,7 @@ impl<E: PairingEngine, FS: AlgebraicSponge<E::Fq, 2>, SM: SNARKMode> VarunaSNARK
             ensure!(commitment_randomnesses.iter().all(|r| r == &empty_randomness));
             end_timer!(commit_time);
 
-            circuit_commitments.sort_by(|c1, c2| c1.label().cmp(c2.label()));
```

Comment:
Convince me this is safe. We use the `circuit_commitments` ordering in `Poseidon` for hashing.

Reply:
this was just a drive-by, so I'm happy to drop this one commit; however, changing the sort from the default (stable) to unstable only results in a difference if the order of the _equal_ elements matters, which is rare